src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -51,31 +69,67 @@ public void Dispose()
             // nop
         }
 
-        public Task StartAsync(CancellationToken cancellationToken)
+        public async Task StartAsync(CancellationToken cancellationToken)
         {
-            _poll = _folderSource.CreateFileWatcher(_fileWatcherType, OnFileWatcher, nextItem: null, pollIntervalInSeconds: _pollIntervalInSeconds > 0 ? _pollIntervalInSeconds : DefaultPollIntervalInSeconds);
+            CloudStorageAccount account;
+            if (!CloudStorageAccount.TryParse(this._config.StorageConnectionString, out account))
+            {
+                throw new ArgumentException(string.Format("The default storage account connection string is invalid: {0}", this._config.StorageConnectionString));
+            }
+
+            CloudBlobClient blobClient = account.CreateCloudBlobClient();
+
+            string apiHubBlobDirectoryPath = string.Format(ApiHubBlobDirectoryPathTemplate, this._siteName);
+            this._apiHubBlobDirectory = blobClient.GetContainerReference(HostContainerName).GetDirectoryReference(apiHubBlobDirectoryPath);
+
+            // Need to start polling from the point where it was left off. if this is the first time then lastPoll will be null.
+            var lastPoll = await GetLastPollStatusAsync();
+
+            Uri nextUri = null;        
+            Uri.TryCreate(lastPoll, UriKind.Absolute, out nextUri);
```

Comment:
You're using TryCreate presumably because this might fail? If so, then recommend handling if this returns false